CHANGELOG.md

```diff
@@ -8,6 +8,7 @@
 - `[expect]` Improve report when mock-spy matcher fails, part 2 ([#8649](https://github.com/facebook/jest/pull/8649))
 - `[expect]` Improve report when mock-spy matcher fails, part 3 ([#8697](https://github.com/facebook/jest/pull/8697))
 - `[expect]` Improve report when mock-spy matcher fails, part 4 ([#8710](https://github.com/facebook/jest/pull/8710))
+- `[expect]` Improve report when mock-spy matcher fails, part 5 ([#8747](https://github.com/facebook/jest/pull/8747))
```

Comment:
Maybe this should be distinct from the others, "Improve error message when trying to use a Jasmine spy for `toHaveReturned`"